### PR TITLE
Adding strided put completion semantics

### PR DESCRIPTION
**Pull request by manjugv**
Clarify shmem_iput completion semantics

### CODE REVIEW

File: content/shmem_iput.tex [hunk 1]
```diff
@@ -69,9 +69,8 @@
     on specified remote \ac{PE}. Both strides, \VAR{dst} and \VAR{sst}, must be
     greater than or equal to \CONST{1}. The routines return when the data has
     been copied out of the \VAR{source} array on the local \ac{PE} but not
-    necessarily before the data has been delivered to the remote data object.
-    If the context handle \VAR{ctx} does not correspond to a valid context,
-    the behavior is undefined.
+    necessarily before the data has been delivered to the remote data object. Use
+    \FUNC{shmem\_quiet} to force completion of all \FUNC{iput} transfers.
```
Comment (jdinan) [inline content/shmem_iput.tex]: I don't see a statement like this for `put` operations.  I wonder if the change below to quiet is sufficient?  Also, looks like we lost a sentence as part of this change.

Comment (manjugv) [reply content/shmem_iput.tex]: @jdinan I was trying to keep consistent with shmem_p. Also, we have a similar sentence in nonblocking puts. I did not realize that we did not have a sentence like that in shmem_put. I added one sentence to shmem_put.

**Comment by manjugv** [comment]
May F2F: Update the text to include put_with_signal interface.

File: content/shmem_iput.tex [hunk 2]
```diff
@@ -70,8 +70,10 @@
     greater than or equal to \CONST{1}. The routines return when the data has
     been copied out of the \VAR{source} array on the local \ac{PE} but not
     necessarily before the data has been delivered to the remote data object.
-    If the context handle \VAR{ctx} does not correspond to a valid context,
-    the behavior is undefined.
+	The routines are considered complete after a subsequent call to
+	\FUNC{shmem\_quiet}.
+	If the context handle \VAR{ctx} does not correspond to a valid context,
+     the behavior is undefined.
 }
```
Comment (BryantLam) [inline content/shmem_iput.tex]: (Notwithstanding the metadiscussion regarding memory model, needing definitions for completion, etc.)

These two lines at 75-76 are marked as a change when there doesn't look like there's any text that has changed. I think there's a whitespace indentation that is messing up the diff. Please fix so these lines don't appear in the git diff.

File: content/shmem_quiet.tex [hunk 3]
```diff
@@ -22,7 +22,7 @@
 \end{apiarguments}
 
 \apidescription{ 
-    The \FUNC{shmem\_quiet} routine ensures completion of \PUT{}, \ac{AMO},
+    The \FUNC{shmem\_quiet} routine ensures completion of \PUT{}, strided \PUT{}, \ac{AMO},
```
Comment (BryantLam) [inline content/shmem_quiet.tex]: I think the general feedback was to make a broader change to define Put operations as block put, elemental put, and strided put to simplify this sentence and related sentences throughout the text.

**Comment by manjugv** [comment]
May F2F: Change the text in the document so PUT refers to Put, P, and strided Put routines.

**Comment by jdinan** [comment]
Superseded by #278

**Pull request closed without merging**